core-tests/shared/src/test/scala/zio/ScheduleSpec.scala

```diff
@@ -319,6 +321,23 @@ object ScheduleSpec extends ZIOBaseSpec {
           (Schedule.stop || (Schedule.spaced(2.seconds) && Schedule.stop)) >>> testElapsed
         )(List.fill(5)(()))
       )(equalTo(List(Duration.Zero)))
+    },
+    testM("perform log for each recurrence of effect") {
```

Comment:
Thanks for contributing! You can create a `Ref`, then update the `Ref` from within `onDecision`, and get the `Ref` at the end of your test to verify that this method behaves correctly.